zerver/lib/avatar.py

```diff
@@ -51,7 +52,8 @@ def _get_unversioned_avatar_url(avatar_source, email=None, realm_id=None,
             # If we can, avoid doing a database query to fetch user_profile
             hash_key = user_avatar_path_from_ids(user_profile_id, realm_id)
         else:
-            user_profile = get_user_profile_by_email(email)
+            realm = Realm.objects.get(id=realm_id)
+            user_profile = get_user(email, realm)
             hash_key = user_avatar_path(user_profile)
         return upload_backend.get_avatar_url(hash_key, medium=medium)
```

Comment:
We should now be asserting that `realm_id is not None` in this code path, right?